packages/react/src/refs.tsx

```diff
@@ -192,8 +200,26 @@ export function Template({
     const release = refStateManager.current.activate()
     return () => release()
   }, [])
+
   const rootDataState = resolveRef(dataValues, path)
   return (
-    <BaseTemplate components={components} path={path} dataState={rootDataState} onEvent={onEvent} />
+    <BaseTemplate
+      components={components}
+      path={path}
+      dataState={rootDataState}
+      onAction={onAction}
+      onEvent={async (event) => {
+        const res = await onEvent(event)
```

Comment:
This could always call `dataSource.onEvent` to make sure it always work, but also expose the "onEvent" prop in form of side-effect

Reply:
```ts
onEvent?.(event)
const res = await dataSource.sendEvent(event)
```